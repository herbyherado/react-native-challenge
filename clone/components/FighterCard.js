import React, { Component } from 'react';
import { Text, View, TouchableHighlight, Image, StyleSheet } from 'react-native'


class FighterCard extends Component {
  render() {
    let fighter = this.props.fighter
    console.log(fighter)
    return (
      <TouchableHighlight onPress={ this.props.press }>
        <View>
          <Image style={styles.img} source={{ uri: fighter.profile_image }} />
          <Text> Name: { `${fighter.first_name} ${fighter.last_name}` } </Text>
          <Text> Wins: { fighter.wins } </Text>
          <Text> Losses: { fighter.losses } </Text>
        </View>
      </TouchableHighlight>
    )
  }
};

const styles = StyleSheet.create({
  img: {
    width: 150,
    height: 150,
    alignSelf: 'center'
  }
})
export default FighterCard