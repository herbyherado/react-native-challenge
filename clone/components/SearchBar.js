import React, { Component } from 'react';
import { View, TextInput, StyleSheet } from 'react-native'

class SearchBar extends Component {
  render() {
    return (
      <View>
        <TextInput
          style={ styles.bar }
          placeholder='Search...'
          defaultValue={ this.props.defaultVal }
          onChangeText={ this.props.changeText }
        />
      </View>
    )
  }
};

const styles = StyleSheet.create({
  bar: {
    height: 30, 
    width: 200, 
    borderColor: 'gray', 
    borderBottomWidth: 1, 
    marginTop: 20, 
    marginBottom: 30, 
    paddingBottom: 5
  }
})

export default SearchBar