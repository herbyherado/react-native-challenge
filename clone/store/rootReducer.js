import { combineReducers } from 'redux'
import Fighter from './fighters/fighter.reducer'
import Octagirl from './octagirls/octagirls.reducer'

const reducers = combineReducers({
  fighters: Fighter,
  octagirls: Octagirl
})

export default reducers