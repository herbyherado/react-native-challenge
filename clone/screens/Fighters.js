import React from 'react';
import { StyleSheet, Text, View, FlatList, Image, TextInput, TouchableHighlight } from 'react-native';
import { TabNavigator } from 'react-navigation'
import { connect } from 'react-redux'
import { searchFighters } from '../store/fighters/fighter.actions'
import FighterCard from '../components/FighterCard'
import SearchBar from '../components/SearchBar'

class Fighters extends React.Component { 
 
  handleChange = (e) => {
    this.props.updateSearch(e)
  }
  
  goToDetail = (fighter) => {
    this.props.navigation.navigate('FighterStats', { title: `${fighter.first_name} ${fighter.last_name}`, data: fighter })
  }

  renderItem = ({item}) => (
    <FighterCard 
      fighter={item}
      press={ () => this.goToDetail(item) } 
      key={`fighter-${item.id}`}
    />
  )
  
  keyExtractor = (item, index) => `fighter-${item.id}`

  render() {
    let regex = new RegExp(this.props.fighters.query, 'gi')
    let arr = this.props.fighters.fighters.filter(fighter => regex.test(fighter.first_name) || regex.test(fighter.last_name) )
    return (
      <View style={ styles.body }>
        <View style={ styles.container }>
          <SearchBar changeText={ this.handleChange } defaultVal={ this.props.fighters.query }/>
          { arr.length ? (
              <FlatList 
                data={ arr }
                renderItem={ this.renderItem }
                keyExtractor={ this.keyExtractor }
              />
            ) : (
              <Text style={{ fontSize: 25 }}> No result found for '{ this.props.fighters.query }' </Text>
            )
          }
        </View>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 2,
    backgroundColor: '#fff',
    alignItems: 'center',
  },
  list: {
    fontSize: 20,
    paddingTop: 5,
    paddingBottom: 5
  },
  body: {
    flex: 1,
    backgroundColor: '#fff'
  }
});

function mapStateToProps (state) {
  return {
    fighters: state.fighters
  }
}

function mapDispatchToProps (dispatch) {
  return {
    updateSearch: (payload) => dispatch(searchFighters(payload))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Fighters)