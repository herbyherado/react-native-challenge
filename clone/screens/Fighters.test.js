import React, { Component } from 'react'
import enzyme, { shallow, mount } from 'enzyme'
import { Text, View, Image } from 'react-native'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'

import Fighters from './Fighters'
import SearchBar from '../components/SearchBar'
import FighterCard from '../components/FighterCard'

import store from '../store/store'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import { searchFighters, getFighters } from '../store/fighters/fighter.actions'

const mockStore = configureStore()
const initialState = {
  fighters: [],
  query: ''
}
enzyme.configure({ adapter: new Adapter()})

let mstore
let wrapper

beforeEach(() => {
  mstore = mockStore(initialState)
  wrapper = shallow(<Provider store={store}><Fighters/></Provider>)
})

describe('Snapshot test <Fighters/>',()=>{
  it('capturing Snapshot of Fighters', () => {
      const renderedValue =  renderer.create(<Provider store={store}><Fighters/></Provider>).toJSON()
      expect(renderedValue).toMatchSnapshot();
  })
})

describe('<Fighters/>', () => {
  it('should render <Fighters/>', () => {
    expect(wrapper).toBeDefined()
    expect(wrapper.length).toEqual(1)
  })
  it('should have components', () => {
    expect(wrapper.find('View')).toBeDefined()
    expect(wrapper.find('SearchBar')).toBeDefined()
    expect(wrapper.find('Text')).toBeDefined()
    expect(wrapper.find('FighterCard')).toBeDefined()
  })
})

// it('+++ check action on dispatching ', () => {
//   let action
//   // store.dispatch(searchFighters('hello'))
//   action = mstore.getActions()
//   console.log(action)
//   // expect(wrapper.find(Fighters).prop('query')).toBe('hello')
//   const q = searchFighters('hello')
//   expect(q).toEqual({type:"FIGHTERS_QUERY", query: 'hello'})
//   // expect(action.type).toBe("FIGHTERS_QUERY")
// });