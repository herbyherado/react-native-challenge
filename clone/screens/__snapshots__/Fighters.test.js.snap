// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test <Fighters/> capturing Snapshot of Fighters 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flex": 2,
      }
    }
  >
    <View>
      <TextInput
        allowFontScaling={true}
        defaultValue=""
        onChangeText={[Function]}
        placeholder="Search..."
        style={
          Object {
            "borderBottomWidth": 1,
            "borderColor": "gray",
            "height": 30,
            "marginBottom": 30,
            "marginTop": 20,
            "paddingBottom": 5,
            "width": 200,
          }
        }
      />
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 25,
        }
      }
    >
       No result found for '
      
      ' 
    </Text>
  </View>
</View>
`;
